src/cycles/find-cycles.ts

```diff
@@ -0,0 +1,42 @@
+// This file is a modification of https://github.com/jcoreio/find-cycle/blob/master/directed.js
+// It adds typescript support and get only one startNode (instead of multiple)
+
+export function findCycle<Node>(
+  startNode: Node,
+  getChildrenNodes: (node: Node) => Node[],
```

Comment:
Is the whole `getChildrenNodes` abstraction really useful? It feels like the `findCycles` function has been made unnecessarily generic.

Reply:
:) as you can see in the first line of this file, this was mostly copied
but ok, i'll change it